### DISCUSSION TITLE
Welcome to ractive Discussions!

### DISCUSSION THREAD

**Discussion by evs-chris** [ractivejs/ractive#3354] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by evs-chris** [comment]
Hi, I'm Chris, and I'm the current primary maintainer of ractive.

I came to ractive as a backend dev transitioning into more advanced pages for back-office applications from rails with scriptaculous way back when. At the time, the big frameworks available for frontend were ember and angular 1, both of which didn't really feel right for what I wanted to do, and I happened across ractive, which was refreshingly easy to deal with. I started using ractive as a proof of concept on a project and found that I needed relatively few tweaks to get things running acceptably, so I tried my hand at doing them myself. Rich was kind enough to help me out a bit with my horrible frontend code, and I eventually got to the point that I was a useful contributor.

It turns out that ractive is a bit heavy for widgets embedded at various points in a news article, so Rich went off to make svelte (and rollup), which is doing quite well in the frontend world as a very lightweight and conceptually refreshing compiled framework. Unfortunately, it's not quite as convenient and designer-friendly for large back-office apps as ractive has come to be. Interacting with large data graphs that get cross-linked across varioous components happens to be something that ractive is very good at.

I've evaluated react, vue, svelte, more iterations of angular, and a number of other smaller frameworks/libraries over the years, and I still haven't found anything that fits quite as nicely as ractive. So, that's how I got here and why I'll continue to maintain and try to improve ractive for as long as I'm working on frontend projects.

Thank you for listening to my TED talk, and yes, I know that this is a Wendy's drive-thru.

**Comment by marcalexiei** [comment]
Hello, I'm Marco,

I have been introduced to Ractive when my developer experience was mainly related to jQuery.
In the first weeks I had some difficulties due to the total change of approach to development but after some time I start to enjoy it.

At the moment I'm using Ractive in a big enterprise SPA and it holds up pretty well with some tweaks (template compile during build process, code splitting, lazy loading / unloading, ...)

In the past year I discovered Typescript and I fell in love with it, which made me decide to help out by converting Ractive’s codebase.

Since then, I’ve been helping on some other stuff too, and Chris has trusted me enough to give me write access to the repository 😄

Happy coding!

**Comment by giovannipiller** [comment]
Hi, I’m Giovanni,
I guess I’m mostly known around here for opening issues with “interesting” edge scenarios, and what I affectionally remember as [“the great hunt for memory leaks”](https://github.com/ractivejs/ractive/issues/2899) 😄

I’ve been into frontend development for most of the past 15 years, and found out about Ractive about 5 years ago, while searching for a modern UI library.

The ease of use blew me away. What helped immensely, was being able to approach what was the “cool kids” JS way-of-life (of the day?) at my own pace. Started out by simply including Ractive.js in a page, parsing templates stored somewhere on the HTML page, then gradually introducing stuff like NPM and Webpack to the equation.

What began as a library used in a side-project at my job, became the foundation of a much more complex SPA . The very same mentioned by @marcalexiei since, well, we’re co-workers!

👋
